src/qt/rpcconsole.cpp

```diff
@@ -487,6 +491,69 @@ bool RPCExecutor::commandMatches(const std::vector<std::string>& parsed_command,
 }
 
+/**
+ * @brief Executes the console-only command "generate".
+ * @param parsed_command  A vector of strings with command and parameters, usually generated by RPCExecutor::parseHelper
+ * @param wallet_model  WalletModel to use for the command
+ * @return True if the command was executed, false otherwise.
+ */
+bool RPCExecutor::executeConsoleGenerate(const std::vector<std::string>& parsed_command, const WalletModel* wallet_model)
+{
+    // Catch the console-only generate command with 2 or less parameters before RPC call is executed .
+    if (!parsed_command.empty() && parsed_command[0] == "generate" && parsed_command.size() <= 3) {
+        // Default number of blocks is 1 if missing
+        const std::string nblocks{parsed_command.size() > 1 ? parsed_command[1] : "1"};
+        // An empty maxtries parameters will be modified by generatetoaddress' default value
+        const std::string maxtries{parsed_command.size() > 2 ? parsed_command[2] : ""};
+
+        // Generate address
+        std::string result;
+        std::string address;
+        if (!RPCConsole::RPCExecuteCommandLine(m_node, address, "getnewaddress\n", /*pstrFilteredOut=*/nullptr, wallet_model)) {
+            Q_EMIT reply(RPCConsole::CMD_ERROR, QString("Error: could not generate new address"));
+        } else {
+            if (!RPCConsole::RPCExecuteCommandLine(m_node, result, "generatetoaddress " + nblocks + " " + address + " " + maxtries + "\n", /*pstrFilteredOut=*/nullptr, wallet_model)) {
+                    Q_EMIT reply(RPCConsole::CMD_ERROR, QString("Error: could not generate blocks"));
+            } else {
+                    std::string answer = "{\n  \"address\": \"" + address + "\",\n  \"blocks\": " + result + "\n}";
+                    Q_EMIT reply(RPCConsole::CMD_REPLY, QString::fromStdString("\n" + answer + "\n\n"));
+            }
+        }
+        return true;
+    } else {
+        // Default to "help generate" when wrong number of parameters are given
+        if (!parsed_command.empty() && parsed_command[0] == "generate" && parsed_command.size() > 3) {
+            return executeConsoleHelpGenerate(parsed_command, wallet_model);
+        } else {
+            return false;
+        }
+    }
+}
+
+/**
+ * @brief  Executes the console-only command "help generate".
+ * @param parsed_command  A vector of strings with command and parameters, usually generated by RPCExecutor::parseHelper
+ * @param wallet_model  WalletModel to use for the command
+ * @return  True if the command was executed, false otherwise.
+ */
+bool RPCExecutor::executeConsoleHelpGenerate(const std::vector<std::string>& parsed_command, const WalletModel* wallet_model)
+{
+    // Catch the console-only "help generate" command when requested, or when "generate is called with wrong parameters.
+    if ((parsed_command.size() >= 2 && parsed_command[0] == "help" && parsed_command[1] == "generate") || (parsed_command.size() > 3 && parsed_command[0] == "generate")) {
```

Comment:
In 411b1da4:

Similar to the one above, The `parsed_command[0] == "help" && parsed_command[1] == "generate"` statement is not needed. The code wouldn't reach this point if the command wouldn't be a `help generate` command.

Reply:
It would if -> Default to "help generate" when wrong number of parameters are given (line 524), but it's the same case anyway, it's the condition after the`or`operator,

Reply:
Modified.